src/Voron/Data/Tables/Table.cs

```diff
@@ -36,8 +36,8 @@ public sealed unsafe class Table : IDisposable
         private FixedSizeTree _inactiveSections;
         private FixedSizeTree _activeCandidateSection;
 
-        private readonly Dictionary<Slice, Tree> _treesBySliceCache = new(SliceStructComparer.Instance);
-        private readonly Dictionary<Slice, Dictionary<Slice, FixedSizeTree>> _fixedSizeTreeCache = new(SliceStructComparer.Instance);
+        private Dictionary<Slice, Tree> _treesBySliceCache;
+        private Dictionary<Slice, Dictionary<Slice, FixedSizeTree>> _fixedSizeTreeCache;
```

Comment:
Do we want to reduce the allocations and create those only if really needed?

Reply:
There is no scenario that I could find that those are not created anyways.

Reply:
So if they are always created then what's the reason of making this change?

Reply:
Because I messed up the comment, dont know how I wrote the exact opposite of what I did. There are scenarios where only one of them would be used, that's why I remove the forced creation. They would only be created on calling `.GetTree()` and `.GetFixedSizeTree()`

Reply:
Thanks. That makes sense now.